### ISSUE TITLE
Monitor Task deletes previous comments

### ISSUE THREAD

**Issue by a-roberts** [tektoncd/experimental#404]
I'm seeing this with our 0.3.0 OpenShift extension release yaml

- Install Tekton Pipelines and Tekton Triggers through openshift-pipelines 0.8.1 into OCP 4.2
- Create a webhook
- Create a pull request, add some comments
- You can either push to your branch for the PR or reopen your PR
- Wait for your PipelineRun to fail
- Refresh the page
- Observe your previous comments are now gone

Logs
```
adams-mbp:config aroberts$ k logs monitor-taskrun-cc5zr-pod-2835bd  --all-containers
{"level":"warn","ts":1576164060.6265314,"logger":"fallback-logger","caller":"logging/config.go:69","msg":"Fetch GitHub commit ID from kodata failed: \"KO_DATA_PATH\" does not exist or is empty"}
{"level":"info","ts":1576164060.6273847,"logger":"fallback-logger","caller":"creds-init/main.go:40","msg":"Credentials initialized."}
{"level":"warn","ts":1576164070.8832994,"logger":"fallback-logger","caller":"logging/config.go:69","msg":"Fetch GitHub commit ID from kodata failed: \"KO_DATA_PATH\" does not exist or is empty"}
{"level":"info","ts":1576164070.8966467,"logger":"fallback-logger","caller":"bash/main.go:64","msg":"Successfully executed command \"sh -c mkdir -p /workspace/output/pull-request\"; output "}
{"level":"warn","ts":1576164071.6034527,"logger":"fallback-logger","caller":"logging/config.go:69","msg":"Fetch GitHub commit ID from kodata failed: \"KO_DATA_PATH\" does not exist or is empty"}
{"level":"info","ts":1576164071.603522,"logger":"fallback-logger","caller":"pullrequest-init/main.go:44","msg":"RUNNING DOWNLOAD!"}
{"level":"info","ts":1576164072.3802173,"logger":"fallback-logger","caller":"pullrequest-init/github.go:213","msg":"Writing comment 16573467 to file: /workspace/pull-request/github/comments/16573467.json"}
Setting status to pending with URL : https://github.ibm.com/api/v3/repos/AROBERTS/devops-world-example/statuses/f05b27e2459274f089ddf71a4ee1d5e68878b087
<Response [201]>
Checking pipelinerun buildah-pipeline-run-2575q in namespace tekton-pipelines
Failed - pipelinerun buildah-pipeline-run-2575q in namespace tekton-pipelines
Set details url to: http://tekton-dashboard-tekton-pipelines.apps.hogans.os.fyre.ibm.com/#/pipelineruns
{"level":"warn","ts":1576164084.3279767,"logger":"fallback-logger","caller":"logging/config.go:69","msg":"Fetch GitHub commit ID from kodata failed: \"KO_DATA_PATH\" does not exist or is empty"}
{"level":"info","ts":1576164084.3280566,"logger":"fallback-logger","caller":"pullrequest-init/main.go:54","msg":"RUNNING UPLOAD!"}
{"level":"info","ts":1576164084.328551,"logger":"fallback-logger","caller":"pullrequest-init/github.go:243","msg":"Syncing path: &{Type: ID:0 Head:0xc4202b84e0 Base:0xc4202b8420 Statuses:[0xc4202be180] Comments:[0xc4200bfd40] Labels:[] Raw: RawStatus:} to pr 32"}
{"level":"info","ts":1576164085.0560188,"logger":"fallback-logger","caller":"pullrequest-init/github.go:281","msg":"Current labels: map[]"}
{"level":"info","ts":1576164085.0561073,"logger":"fallback-logger","caller":"pullrequest-init/github.go:292","msg":"Creating labels [] for PR 32"}
{"level":"info","ts":1576164085.3166075,"logger":"fallback-logger","caller":"pullrequest-init/github.go:312","msg":"Setting comments for PR 32 to: [0xc4200bfd40]"}
{"level":"info","ts":1576164085.4383652,"logger":"fallback-logger","caller":"pullrequest-init/github.go:361","msg":"Deleting comment 16573467 for PR 32"}
{"level":"info","ts":1576164085.6606717,"logger":"fallback-logger","caller":"pullrequest-init/github.go:391","msg":"Creating comment ## Tekton Status Report \n\nStatus | Pipeline | PipelineRun | Namespace\n:----- | :------- | :--------------- | :--------\n[**Failed**](http://tekton-dashboard-tekton-pipelines.apps.hogans.os.fyre.ibm.com/#/namespaces/tekton-pipelines/pipelineruns/buildah-pipeline-run-2575q) | buildah-pipeline | buildah-pipeline-run-2575q | tekton-pipelines for PR 32"}
```

Specific yaml I applied is from https://github.com/tektoncd/dashboard/releases/tag/v0.3.0

Notice the deleting comment part 👀 

![image](https://user-images.githubusercontent.com/12814972/70726317-d1c2ef00-1cf5-11ea-98ba-73115e25a804.png)

before ^^

Close and reopen the PullRequest:

![image](https://user-images.githubusercontent.com/12814972/70726355-e0110b00-1cf5-11ea-8ce5-96f92f950014.png)

Refresh

![image](https://user-images.githubusercontent.com/12814972/70726404-eacba000-1cf5-11ea-8d0b-a1367d094cd5.png)

**Comment by a-roberts** [comment]
Have a fix for this

```
      comment = ("## Tekton Status Report \n\n"
                 "Status | Pipeline | PipelineRun | Namespace\n"
                 ":----- | :------- | :--------------- | :--------\n"
                 ) + "\n".join(results)
      shutil.copytree("/workspace/pull-request/comments","/workspace/output/pull-request/comments")
      handle = open("/workspace/output/pull-request/comments/newcomment.json", 'w')
      handle.write(comment)
      handle.close()
```

testing with no existing comments now

**Comment by a-roberts** [comment]
^^ all is well, adding the fix into the release assets and 0.3.0 branch now

**Comment by a-roberts** [comment]
This was merged a while back so closing
